cpp/include/cudf/detail/null_mask.cuh

```diff
@@ -88,15 +92,8 @@ __global__ void offset_bitmask_binop(Binop op,
 
     destination[destination_word_index] = destination_word;
     thread_count += __popc(destination_word);
-  }
-
-  // Subtract any slack bits from the last word
-  if (tid == 0) {
-    size_type const last_bit_index = source_size_bits - 1;
-    size_type const num_slack_bits = word_size - (last_bit_index % word_size) - 1;
-    if (num_slack_bits > 0) {
-      size_type const word_index = cudf::word_index(last_bit_index);
-      thread_count -= __popc(destination[word_index] & set_most_significant_bits(num_slack_bits));
+    if (destination_word_index == last_word_index) {
```

Comment:
Would it be simpler / safer to count _only_ the bits we know to be non-slack, rather than counting all bits and subtracting the slack bits?
```
thread_count += __popc(destination_word_index == last_word_index
                           ? destination_word & set_least_significant_bits(
                                                    word_size - num_slack_bits)
                           : destination_word);
```

(I would refactor `num_slack_bits` to be `num_bits_in_last_word` to eliminate the subtraction)

Reply:
It would also look a bit more like the implementation here: https://github.com/rapidsai/cudf/blob/ebc68df255b9f304ee19b969645a10d5ab9f8bea/cpp/src/bitmask/null_mask.cu#L119-L121

Reply:
To clarify my pair of suggestions: maybe we should write this above:
```cpp
auto const num_bits_in_last_word = cudf::detail::intra_word_index(last_bit_index);
```
and then here, write:
```cpp
thread_count += __popc(destination_word_index == last_word_index ? destination_word & set_least_significant_bits(num_bits_in_last_word) : destination_word);
```

Reply:
Opted to mask the word itself to keep it to one if-statement on no ternary.

Reply:
This change is even better. Not only does it fix the count, it also zeros the slack bits in the result. 👍